src/main/java/org/wso2/carbon/apimgt/securityenforcer/PingAISecurityHandler.java

```diff
@@ -0,0 +1,331 @@
+import org.apache.synapse.Mediator;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.AbstractHandler;
+import org.apache.synapse.rest.RESTConstants;
+import org.apache.synapse.transport.passthru.PassThroughConstants;
+import org.apache.synapse.transport.passthru.util.RelayUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.wso2.carbon.apimgt.gateway.handlers.Utils;
+import org.wso2.carbon.apimgt.gateway.handlers.security.AuthenticationContext;
+import org.wso2.carbon.apimgt.securityenforcer.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.securityenforcer.utils.AISecurityException;
+import org.wso2.carbon.apimgt.securityenforcer.utils.AISecurityHandlerConstants;
+import org.wso2.carbon.apimgt.securityenforcer.utils.SecurityUtils;
+
+import java.util.Date;
+import java.util.TreeMap;
+
+/**
+ * This class is Handling the Ping Identity analysis. This class will use inside each API as securityenforcer.
+ * It will fetch some of meta data from incoming message and send them to PING API Security Enforcer.
+ */
+public class PingAISecurityHandler extends AbstractHandler {
+
+    private static final Log log = LogFactory.getLog(PingAISecurityHandler.class);
+
+    public PingAISecurityHandler() {
+        log.debug("Ping AI Security Handler initialized");
+    }
+
+    /**
+     * This method will handle the request. For every request gateway receives, this is method will invoke first for this handler
+     */
+    @Override
+    public boolean handleRequest(MessageContext messageContext) {
+
+        long handleRequestStartTime = System.nanoTime();
+        String requestCorrelationID = SecurityUtils.getAndSetCorrelationID(messageContext);
+        try {
+            if (authenticate(messageContext, requestCorrelationID)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Handle Request Time for the request " + requestCorrelationID + " is " + (System.nanoTime()
+                                    - handleRequestStartTime) + " Nano seconds");
+                }
+                SecurityUtils.updateLatency(System.nanoTime() - handleRequestStartTime, messageContext);
+                return true;
+            }
+        } catch (AISecurityException e) {
+            if (log.isDebugEnabled()) {
+                long difference = System.nanoTime() - handleRequestStartTime;
+                String messageDetails = logMessageDetails(messageContext);
+                log.debug("Request " + requestCorrelationID + " failed. " + messageDetails + ", elapsedTimeInNano"
+                        + difference);
+            }
+            handleAuthFailure(messageContext, e);
+        } finally {
+            SecurityUtils.updateLatency(System.nanoTime() - handleRequestStartTime, messageContext);
+        }
+        return false;
```

Comment:
You can just have a single return true statement at the end in place of having separate return true and return false statements